sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java

```diff
@@ -239,6 +245,12 @@ public void waitForSuccess(Duration duration) throws IOException {
 
     String result = pollForResultForDuration(resultSubscriptionPath, duration);
 
+    try {
+      pubsub.deleteSubscription(resultSubscriptionPath);
+    } catch (IOException e) {
+      LOG.warn(String.format("Leaked PubSub subscription '%s'", resultSubscriptionPath));
```

Comment:
nit: make error will make it easier to be discovered.

Reply:
Good idea, done